Blaise.Tests.Helpers/Browser/BrowserHelper.cs

```diff
@@ -79,5 +90,100 @@ private static ChromeDriver CreateChromeDriver()
 
             return new ChromeDriver(BrowserConfigurationHelper.ChromeDriver, chromeOptions);
         }
+
+        //AP changes - 20-03-2023
+        public static IWebElement WaitForElement(By by, bool checkIfClickable = false, int maxAttempts = 50)
+        {
+            IWebElement element = null;
+            var attempts = 0;
+
+            while (element == null && attempts < maxAttempts)
+            {
+                try
+                {
+                    element = _browser.FindElement(by);
+                }
+                catch (Exception)
+                {
+                    /*Ignored - continue polling*/
+                }
+
+                if (element != null && IsElementVisible(element))
+                {
+                    if (checkIfClickable)
+                    {
+                        if (IsElementClickable(element))
+                        {
+                            break;
+                        }
+                    }
+                    else
+                    {
+                        break;
+                    }
+                }
+                else
+                {
+                    System.Threading.Thread.Sleep(500);
+                }
+
+                attempts++;
+            }
+
+            if (element == null)
+            {
+                var f = "";
+                //Log this error
+                //   throw new NoSuchElementException($"Could not find element by {by} after {maxAttempts} attempts.");
+            }
+
+            return element;
+        }
+
+
+        private static bool IsElementVisible(IWebElement element)
+        {
+            return element.Displayed && element.Enabled;
+        }
+
+        private static bool IsElementClickable(IWebElement element)
+        {
+            try
+            {
+                if (IsElementVisible(element))
+                {
+                    element.Click();
```

Comment:
I was surprised to see this here, I wouldn't expected "is clickable" to actually click. I'd say it should either just return `true` or `false`, or be renamed to something like "click element if clickable"?

Reply:
Refactored out